new_csaf/csaf/test/base.py

```diff
@@ -1,4 +1,234 @@
-class CsafTest:
-    pass
+from os import stat
+import csaf
+from f16lib.acas import AcasScenario, AcasShieldScenario, AcasScenarioViewer
 
+from collections.abc import Callable
 
+from abc import ABC, abstractmethod
+
+import numpy as np
+import typing
+import GPyOpt
+import GPy
+
+class Optimizer(ABC):
+    """
+    Generic optimizer
+    """
+    @abstractmethod
+    def prepare(self,
+                configuration_space: dict,
+                obj_func: Callable[[[float]], float]):
+        pass
+
+    @abstractmethod
+    def run_optimization(self,
+                         max_iter: int,
+                         max_time: float,
+                         tolerance: float,
+                         verbose: bool = False):
+        pass
+
+class Scenario(ABC):
+    """
+    - system_under_test is the system being tested
+    - configuration_space is a dictionary specifying
+    the configuration space of the problem
+    """
+    configuration_space: dict = {}
```

Comment:
Configuration space would need to a `NamedTuple` or the like. Spaces can be indexed, and so need ordering. Their primitive descriptions can be named.